tests/test_chebyshev.py

```diff
@@ -0,0 +1,135 @@
+# See COPYRIGHT file at the top of the source tree.
+#
+# This file is part of fgcmcal.
+#
+# Developed for the LSST Data Management System.
+# This product includes software developed by the LSST Project
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+"""Test the fgcm and fgcmcal (stack) chebyshev polynomial code.
+
+Run small tests to ensure that chebyshev implementations are equivalent.
+"""
+
+import matplotlib
+matplotlib.use("Agg")  # noqa E402
+
+import unittest
+import numpy as np
+
+import lsst.pex.exceptions
+import lsst.afw.math as afwMath
+import lsst.geom
+
+import fgcm
+
+
+class FgcmChebyshevTest(lsst.utils.tests.TestCase):
+    def setUp(self):
+        self.xSize = 2000
+        self.ySize = 4000
+        self.nStar = 1000
+
+        self.order = 2
+        self.pars = np.zeros((self.order + 1, self.order + 1))
+        self.pars[0, 0] = 1.0
+        self.pars[0, 1] = 3e-3
+        self.pars[0, 2] = 1e-5
+        self.pars[1, 0] = 5e-3
+        self.pars[1, 1] = 7e-6
+        self.pars[1, 2] = 0.0
+        self.pars[2, 0] = 3e-5
+        self.pars[2, 1] = 0.0
+        self.pars[2, 2] = 0.0
+
+        self.order2 = 1
+        self.pars2 = np.zeros((self.order2 + 1, self.order2 + 1))
+        self.pars2[0, 0] = 0.98
+        self.pars2[0, 1] = 5e-3
+        self.pars2[1, 0] = 2e-2
+
+    def test_chebyshev_evaluate(self):
+        # Set numpy seed for stability
+        np.random.seed(seed=1000)
+
+        xPos = self.xSize * np.random.rand(self.nStar)
+        yPos = self.ySize * np.random.rand(self.nStar)
+
+        bbox = lsst.geom.Box2I(lsst.geom.Point2I(0, 0),
+                               lsst.geom.Point2I(self.xSize - 1, self.ySize - 1))
+
+        # Compute the chebyshev values using the fgcm code
+        fgcmField = fgcm.fgcmUtilities.Cheb2dField(self.xSize, self.ySize,
+                                                   self.pars)
+        fgcmValues = fgcmField.evaluate(xPos, yPos)
+
+        # Compute the chebyshev values using the afw code
+        field = afwMath.ChebyshevBoundedField(bbox, self.pars)
```

Comment:
Just so I understand this - the fcgm Cheb2dField pre-dates (and/or was developed independently of) afw's ChebyshevBoundedField but should return nearly-identical results for any input?

Reply:
Yes, and I added some documentation to the top of the `test_chebyshev.py` file to make this clearer.